### PR TITLE
profiler: use only one service version value

### PR DESCRIPTION
**Pull request by nsrip-dd**
### What does this PR do?

Users can set the version tag for profiles in three ways: via
`WithVersion`, the `DD_VERSION` environment variable, or via `WithTags`. A
service should only have one version tag at a time for profile
comparison with a "recent" version to work properly in the UI. Today,
all of these options can be used at once and the profiles for a service
can have multiple version tags.

To be consistent with the tracer, we should supply a single version and
it should be configured with the following priority order: first,
`WithVersion` is used. If that option is not set, then `DD_VERSION` will be
used. If either of those are used, then they get priority over any
version tags supplied via `WithTags`. That is, version tags are removed if
the version is explicity set.

### Reviewer's Checklist

- [x] Changed code has unit tests for its functionality at or near 100% coverage.
- [x] There is a benchmark for any new code, or changes to existing code.
- [x] If this interacts with the agent in a new way, a system test has been added.

For Datadog employees:

- [ ] If this PR touches code that handles credentials of any kind, such as Datadog API keys, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: profiler/profiler.go [hunk 1]
```diff
@@ -192,6 +194,21 @@ func newProfiler(opts ...Option) (*profiler, error) {
 	if cfg.logStartup {
 		logStartup(cfg)
 	}
+	var tags []string
+	for _, tag := range cfg.tags.Slice() {
+		// If the user configured a tag via DD_VERSION or WithVersion,
+		// override any version tags the user provided via WithTags,
+		// since having more than one version tag breaks the comparison
+		// UI
+		if cfg.version != "" && strings.HasPrefix(tag, "version:") {
```
Comment (felixge) [inline profiler/profiler.go:203]: Q: Do we need to worry about uppercase here?

File: profiler/profiler.go [hunk 2]
```diff
@@ -192,6 +194,21 @@ func newProfiler(opts ...Option) (*profiler, error) {
 	if cfg.logStartup {
 		logStartup(cfg)
 	}
+	var tags []string
+	for _, tag := range cfg.tags.Slice() {
+		// If the user configured a tag via DD_VERSION or WithVersion,
+		// override any version tags the user provided via WithTags,
+		// since having more than one version tag breaks the comparison
+		// UI
```
Comment (felixge) [inline profiler/profiler.go:202]: Q: It seems like we still allow the user to set multiple version tags via `WithTags`. Is that intentional?

**Review by felixge — APPROVED** [review]
LGTM, but see questions.

File: profiler/profiler.go (hunk 1 above)
Comment (nsrip-dd) [reply profiler/profiler.go]: I hadn't considered this. It looks like tags to get normalized somewhere. I tried uploading a profile with `mytag:foo` and `MyTag:bar`, and searching for it only turned up anything with either `mytag:foo` or `mytag:bar` (lower case). So I think it'd be good to be case-insensitive here 👍

File: profiler/profiler.go (hunk 2 above)
Comment (nsrip-dd) [reply profiler/profiler.go]: Ah, good point. We don't normally de-duplicate tags, but I think that'd be user-friendly to do here to match the intention of only ever having one.

**Pull request merged**